### PR TITLE
[android] Refactor media_session_client

### PR DESCRIPTION
**Pull request by zhongqiliang**
1. removed mutex lock in OnMediaSessionStateChanged() in android_media_session_client.cc since it does nothing in the mutex block.
2. removed if (ext_image.src == nullptr) check in OnMediaSessionStateChanged() in media_session_client.cc as media_image.src().c_str() should never return nullptr.
3. earlier return when if (extension_ && extension_->version >= 1) check failed
4. created ext_artwork with vector instead of array.

b/329330347

### CODE REVIEW

**Comment by sideb0ard** [comment]
I'm always a bit reluctant to add a mutex if it can be avoided.
If the issue here is that the reference to MediaMetadataInit goes out of scope, at https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:cobalt/media_session/media_session_client.cc;l=325;drc=dff48f461fe8b269ff79460332dbd5d258262b99;bpv=1;bpt=0 - could we just take a copy of MediaMetadataInit?

**Comment by zhongqiliang** [comment]
From here, https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:cobalt/media_session/media_session.cc;l=99-114?q=p:cobalt%20UpdateMediaSessionState, do we know if UpdateMediaSessionState() can be called on multiple threads? If this is the case, I feel there could be a chance that 1 thread is reading session_state_ fields inside OnMediaSessionStateChanged https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:cobalt/media_session/media_session_client.cc;l=300-350, while the other thread can rewrite session_state_ here https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:cobalt/media_session/media_session_client.cc;l=275-278. Thoughts?

**Comment by sideb0ard** [comment]
> do we know if UpdateMediaSessionState() can be called on multiple threads? 

It doesn't look like it. 
I see it being called twice, once in media_session_client which DCHECKs its on the media_session_->task_runner_, and once in media_session which also checks its on the task_runner_->BelongsToCurrentThread()

**Comment by zhongqiliang** [comment]
I see, thank you. One thing to confirm, "If the issue here is that the reference to MediaMetadataInit goes out of scope", do you meant the ext_image object here, https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:cobalt/media_session/media_session_client.cc;l=333, is it?

File: cobalt/media_session/media_session_client.cc [hunk 1]
```diff
@@ -297,60 +298,57 @@ void MediaSessionClient::UpdateMediaSessionState() {
...
+      session_state.current_playback_position();
+  ext_state.has_position_state = session_state.has_position_state();
+  ext_state.actual_playback_state =
+      ConvertPlaybackState(session_state.actual_playback_state());
+  ConvertMediaSessionActions(session_state.available_actions(),
+                             ext_state.available_actions);
+  std::string album = "";
+  std::string artist = "";
+  std::string title = "";
+  std::vector<CobaltExtensionMediaImage> ext_artwork(artwork_size);
```
[the remark this answers is not in the document]
Comment (zhongqiliang) [reply cobalt/media_session/media_session_client.cc:323]: This vector is just a refactoring, it should be the same as new the array. 

You suggested to create 2 member fields. I thought if we do not make it as a member field, the object will be released at the end of the method, which is after the line of extension_->OnMediaSessionStateChanged(ext_state);
If this is correct, the change won't make it better?

[the remark this answers is not in the document]
Comment (sideb0ard) [reply cobalt/media_session/media_session_client.cc:323]: okay, yeah, scratch that, i see what you've done! and yeah, i agree that should work!

**Review by sideb0ard — APPROVED** [review]

**Pull request closed without merging**